### DISCUSSION TITLE
Clusterdata output is a fixed file name

### DISCUSSION THREAD

**Discussion by timdnewman** [pyg-team/pytorch_geometric#9416] [Ideas]
Hi,

When you use ClusterData on multiple graphs it produces a fixed file name, and the only way not to overwrite them is to put them in separate folders.

Is there any reason we/you/I couldn't add an optional filename parameter

e.g. 
```python
clustered_data = ClusterData(G, num_parts = 500, recursive=False, save_dir=out_path, filename=f"{graphName}_clustered.pt")
```

So that each file could be called something different.

If you made it optional and default it to none it wouldn't break existing code, and the actual change would be simple

```python
if filename is None:
    filename = f'metis_{num_parts}{recursive_str}.pt'
```
Currently the file name is effectively hard coded to ```f'metis_{num_parts}{recursive_str}.pt'```

I've never contributed a change before but that seems simple, but before I got to the effort of cloning and all that I thought I would make sure there isn't some reason I should not bother? If the answer is its too small and someone else will do it when we refactor, its all good. I'm not desperate to get my name in light :)

Thanks,
Tim

**Comment by rusty1s** [comment]
Thanks for the issue. I went ahead and changed `ClusterLoader` accordingly, see https://github.com/pyg-team/pytorch_geometric/pull/9448. Hope this works.

**Reply by timdnewman** [reply]
Thanks!
